### PR TITLE
feat: evaluate c_nonce

### PR DESCRIPTION
**Pull request by blu3beri**
DRAFT: still need to add some tests

## Description

- Added evaluation for the `c_nonce`
- Some documentation fixups
- Expanded the `evaluate_credential_request` tests to be easier to add and tested some edge cases

## Checklist

- [ ] Tests (unit, integration and e2e where relevant)
- [x] Documentation (in docs and within the code)

### CODE REVIEW

File: crates/openid4vci-grpc/proto/access_token.proto [hunk 1]
```diff
@@ -50,6 +50,7 @@ message CreateAccessTokenSuccessResponseResponse {
 
   message Success {
     bytes success_response = 1;
+    string created_at = 2;
```
Comment (TimoGlastra) [inline crates/openid4vci-grpc/proto/access_token.proto:53]: a thought. If we provide c_nonce_expires_in, we could also return c_nonce_expires_at? But that means the user of the library doesn't have control over when the c_nonce expiration will start (they could wait with sending the response object(

File: crates/openid4vci/src/access_token/mod.rs [hunk 2]
```diff
@@ -58,7 +60,18 @@ pub struct AccessTokenSuccessResponse {
 
     /// The minimum amount of time in seconds that the client SHOULD wait between polling requests to the
     /// Token Endpoint in the Pre-Authorized Code Flow. If no value is provided, clients MUST use 5 as the default.
-    pub interval: Option<u64>,
+    pub interval: Option<u32>,
+}
+
+impl Validatable for AccessTokenSuccessResponse {
+    fn validate(&self) -> Result<(), ValidationError> {
+        if self.c_nonce.is_none() && self.c_nonce_expires_in.is_some() {
```
Comment (TimoGlastra) [inline crates/openid4vci/src/access_token/mod.rs:68]: What is c_nonce is some, but c_nonce_expires_in is none? Is that allowed?

**Review by TimoGlastra — APPROVED** [review]

File: crates/openid4vci/src/access_token/mod.rs (hunk 2 above)
Comment (blu3beri) [reply crates/openid4vci/src/access_token/mod.rs:68]: The specification does not specify whether it is required when `c_nonce` is supplied (https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-6.2). So I assume it is allowed for a `c_nonce` which does not expire.
